src/download.rs

```diff
@@ -19,6 +21,17 @@ where
         .send()
         .await?;
 
+    // Return immediately on download failure on the client side.
+    let status = res.status();
+    if status.is_client_error() {
+        match status {
+            StatusCode::FORBIDDEN | StatusCode::NOT_FOUND => {
+                return Err(format!("cannnot fetch remotely with status code {:?}", status).into());
+            }
+            _ => (),
```

Comment:
We shouldn't continue here but also fail for now until we add retry logic.

Reply:
Done